src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +66,792 @@ const ActorInit En_Famos_InitVars = {
+    EnFamosRock* rock;
+    s32 i;
+
+    this->debrisTimer = 40;
+    rock = &this->rocks[0];
+    for (i = 0; i < ARRAY_COUNT(this->rocks); i++, rock++) {
+        randVelDirection = Rand_Next() >> 0x10;
+        randSmaller = (u32)Rand_Next() >> 0x12;
+        randFloat = Rand_ZeroFloat(6.0f) + 7.0f;
+        rock->velocity.x = randFloat * Math_CosS(randSmaller) * Math_SinS(randVelDirection);
+        rock->velocity.y = Math_SinS(randSmaller) * randFloat + 4.5f;
+        rock->velocity.z = randFloat * Math_CosS(randSmaller) * Math_CosS(randVelDirection);
+        rock->rotation.x = Rand_Next() >> 0x10;
+        rock->rotation.y = Rand_Next() >> 0x10;
+        rock->rotation.z = Rand_Next() >> 0x10;
+        rock->pos.x = Math_SinS(randVelDirection) * 20.0f + this->actor.world.pos.x;
+        rock->pos.y = randPlusMinusPoint5Scaled(60.0f) + (this->actor.world.pos.y + 40.0f);
+        rock->pos.z = Math_CosS(randVelDirection) * 20.0f + this->actor.world.pos.z;
+        rock->scale = Rand_ZeroFloat(0.002f) + (2.5f * 0.001f);
+    }
+}
+
+s32 EnFamos_IsPlayerSeen(EnFamos* this, GlobalContext* globalCtx) {
+    if (Player_GetMask(globalCtx) != PLAYER_MASK_STONE &&
+        Actor_XZDistanceToPoint(&GET_PLAYER(globalCtx)->actor, &this->calmPos) < this->agroDistance &&
+        (Actor_IsFacingPlayer(&this->actor, 0x5000))) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void EnFamos_UpdateBobbingHeight(EnFamos* this) {
+    if (this->hoverClk == 0) {
+        this->hoverClk = 30;
+    }
+
+    this->hoverClk--;
+    this->actor.world.pos.y = (Math_SinS(this->hoverClk * 2184) * 10.0f) + this->baseHeight;
+
+    if (ABS_ALT(this->flipRot) > 0x4000) { // is famos upside down
+        func_800B9010(&this->actor, NA_SE_EN_FAMOS_FLOAT_REVERSE - SFX_FLAG);
+    } else {
+        func_800B9010(&this->actor, NA_SE_EN_FAMOS_FLOAT - SFX_FLAG);
+    }
+}
+
+/**
+ * Summary: Checks if emblem has been hit with light arrow.
+ *          Also checks if previously flipped; handles flip status.
+ */
+void EnFamos_UpdateFlipStatus(EnFamos* this) {
+    u8 famosRotTest;
+
+    if (this->emblemCollider.base.acFlags & AC_HIT) { // light arrow collision, flip
+        this->emblemCollider.base.acFlags &= ~AC_HIT;
+        if (this->stableRotation == true) {
+            if (this->animatedMaterialIndex != FAMOS_ANIMATED_MATERIAL_NORMAL) {
+                this->animatedMaterialIndex = FAMOS_ANIMATED_MATERIAL_NORMAL;
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_FAMOS_REVERSE2);
+            } else {
+                this->animatedMaterialIndex = FAMOS_ANIMATED_MATERIAL_FLIPPED;
+                this->flippedTimer = 100;
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_FAMOS_REVERSE1);
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_EYEGOLE_DAMAGE);
+            }
+            this->stableRotation = false;
+        }
+
+    } else if (this->flippedTimer > 0 && --this->flippedTimer == 0) { // check flipped timer
```

Comment:
```suggestion
    } else if ((this->flippedTimer) > 0 && (--this->flippedTimer == 0)) { // check flipped timer
```
I suppose this isn't DECR, but would it match as separate ifs? We try to avoid assignments in conditionals where possible.

Reply:
Since when? Why?

Reply:
Since always, there's only a few in the entire codebase, most of which IIRC are required to match. It makes it clearer if you don't have to figure out if it's using the old or new (and it's bad style to mutate things in conditionals anyway).